pytorch_lightning/plugins/precision/sharded_native_amp.py

```diff
@@ -35,7 +35,7 @@ def __init__(
                 "You have asked for sharded AMP but you have not installed it."
                 " Install `fairscale` using this guide: https://https://github.com/facebookresearch/fairscale"
             )
-        super().__init__(precision, device, scaler=scaler or ShardedGradScaler())
+        super().__init__(precision, device, scaler=ShardedGradScaler() if scaler is None and precision == 16 else None)
```

Comment:
qq:
isn't `ShardedNativeMixedPrecisionPlugin` selected only if precision is 16?